Makefile

```diff
@@ -288,13 +288,17 @@ push:
 	DIMAGE=${IMAGE_ORG}/node-disk-operator-${XC_ARCH} ./build/push;
 	DIMAGE=${IMAGE_ORG}/node-disk-exporter-${XC_ARCH} ./build/push;
 
+.PHONY: controller-gen
+controller-gen:
+	TMP_DIR=$(shell mktemp -d) && cd $$TMP_DIR && go mod init tmp && go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.0 && rm -rf $$TMP_DIR;
+
+
 .PHONY: crds
-crds:
-	touch build/Dockerfile 
-	# Install the binary from https://github.com/operator-framework/operator-sdk/releases/tag/v0.17.0
-	operator-sdk-v0.17.0-x86_64-linux-gnu generate crds
-	rm build/Dockerfile
+crds: controller-gen
+	$(shell which controller-gen) crd:trivialVersions=false,preserveUnknownFields=false paths=./pkg/apis/... output:crd:artifacts:config=deploy/crds
+
```

Comment:
Is this approach made in latest operator SDK also. So that we dont have to make change when operator SDK is updated. cc: @shubham14bajpai @RealHarshThakur

Reply:
Operator SDK internally uses the same approach so using this will help us become independent of it. Using the operator-sdk binary may be challenging when we try upgrading the operator SDK version in future.

Reply:
Is this the approach followed in jiva operator?

Reply:
Yes all other repos the same approach